app-modules/service-management/src/Filament/Resources/ServiceRequestResource/Actions/AddServiceRequestUpdateBulkAction.php

```diff
@@ -0,0 +1,137 @@
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\Filament\Resources\ServiceRequestResource\Actions;
+
+use AidingApp\ServiceManagement\Enums\ServiceRequestUpdateDirection;
+use AidingApp\ServiceManagement\Models\ServiceRequest;
+use AidingApp\ServiceManagement\Models\ServiceRequestUpdate;
+use App\Models\Authenticatable;
+use App\Support\BulkProcessingMachine;
+use Closure;
+use Filament\Forms\Components\Textarea;
+use Filament\Forms\Components\Toggle;
+use Filament\Tables\Actions\BulkAction;
+use Illuminate\Database\Eloquent\Collection;
+
+class AddServiceRequestUpdateBulkAction
+{
+    public static function make(): BulkAction
+    {
+        return BulkAction::make('addServiceRequestUpdate')
+            ->label('Add Request Update')
+            ->icon('heroicon-m-pencil-square')
+            ->modalHeading('Bulk add service request update')
+            ->modalWidth('md')
+            ->modalSubmitActionLabel('Add Update')
+            ->form([
+                Textarea::make('update')
+                    ->label('Update')
+                    ->rows(3)
+                    ->columnSpan('full')
+                    ->required()
+                    ->string(),
+                Toggle::make('internal')
+                    ->label('Internal')
+                    ->rule(['boolean']),
+            ])
+            ->action(function (array $data, Collection $records) {
+                $records->loadMissing(['priority.type.managers', 'serviceRequestUpdates']);
+
+                $user = auth()->user();
+                $isUserSuperAdmin = $user->hasRole(Authenticatable::SUPER_ADMIN_ROLE);
+                $canUserAddServiceRequestUpdate = $user->can('service_request.*.update') && $user->can('service_request_update.create');
```

Comment:
I think you can move thi check down into where the check is used since it is only used in one place.